integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLCastFunctionIT.java

```diff
@@ -0,0 +1,151 @@
+  @Override
+  public void init() throws IOException {
+    super.init();
+    loadIndex(Index.STATE_COUNTRY);
+    loadIndex(Index.STATE_COUNTRY_WITH_NULL);
+  }
+
+  @Test
+  public void testCast() {
+    JSONObject actual =
+        executeQuery(
+            String.format(
+                "source=%s | eval a = cast(age as string) | fields a", TEST_INDEX_STATE_COUNTRY));
+
+    verifySchema(actual, schema("a", "string"));
+
+    verifyDataRows(actual, rows("70"), rows("30"), rows("25"), rows("20"));
+  }
+
+  @Test
+  public void testCastOverriding() {
+    JSONObject actual =
+        executeQuery(
+            String.format(
+                "source=%s | eval age = cast(age as STRING) | fields age",
+                TEST_INDEX_STATE_COUNTRY));
+
+    verifySchema(actual, schema("age", "string"));
+
+    verifyDataRows(actual, rows("70"), rows("30"), rows("25"), rows("20"));
+  }
+
+  @Test
+  public void testCastToUnknownType() {
+    assertThrows(
+        SyntaxCheckException.class,
+        () ->
+            executeQuery(
+                String.format(
+                    "source=%s | eval age = cast(age as VARCHAR) | fields age",
+                    TEST_INDEX_STATE_COUNTRY)));
+  }
+
+  @Test
+  public void testChainedCast() {
+    JSONObject actual =
+        executeQuery(
+            String.format(
+                "source=%s | eval age = cast(concat(cast(age as string), '0') as DOUBLE) | fields"
+                    + " age",
+                TEST_INDEX_STATE_COUNTRY));
+
+    verifySchema(actual, schema("age", "double"));
+
+    verifyDataRows(actual, rows(700.0), rows(300.0), rows(250.0), rows(200.0));
+  }
+
+  @Test
+  public void testCastNullValues() {
+    String actual =
+        execute(
+            String.format(
+                "source=%s | eval a = cast(state as string) | fields a",
+                TEST_INDEX_STATE_COUNTRY_WITH_NULL));
+
+    assertEquals(
+        "{\n"
+            + "  \"schema\": [\n"
+            + "    {\n"
+            + "      \"name\": \"a\",\n"
+            + "      \"type\": \"string\"\n"
+            + "    }\n"
+            + "  ],\n"
+            + "  \"datarows\": [\n"
+            + "    [\n"
+            + "      \"California\"\n"
+            + "    ],\n"
+            + "    [\n"
+            + "      \"New York\"\n"
+            + "    ],\n"
+            + "    [\n"
+            + "      \"Ontario\"\n"
+            + "    ],\n"
+            + "    [\n"
+            + "      \"Quebec\"\n"
+            + "    ],\n"
+            + "    [\n"
+            + "      null\n"
+            + "    ]\n"
+            + "  ],\n"
+            + "  \"total\": 5,\n"
+            + "  \"size\": 5\n"
+            + "}",
+        actual);
+  }
+
+  @Test
+  public void testCastToUnsupportedType() {
+    String actual =
+        execute(
+            String.format(
+                "source=%s | eval a = cast(name as boolean) | fields a", TEST_INDEX_STATE_COUNTRY));
```

Comment:
cast("1" as boolean) should return True?

Reply:
no, only 'true', 'True', 'TRUE' etc could return `True` in cast_to_boolean.
cast("1" as boolean), cast(1 as boolean), cast(2.5 as boolean) are all return `null`.

This seems a breaking change comparing to V2:
cast("1" as boolean) returns false
cast(1 as boolean), cast(2.5 as boolean) return true
cast(0 as boolean) returns false

Reply:
Em, I think this is a bug of Calcite, we can fix it in our side.

Reply:
https://github.com/opensearch-project/sql/issues/3443

Reply:
The latest commit follows the behaviour of SparkSQL. Check above